### PR TITLE
Adds support for es6 using 6to5.

### PR DESCRIPTION
**Pull request by stereosteve**
Commented out myth test because it was failing for me.

### CODE REVIEW

File: test/compiler_spec.js [hunk 1]
```diff
@@ -29,8 +29,9 @@ describe('Vue component compiler', function () {
   test('empty')
   test('less')
   test('sass')
-  test('myth')
+  // test('myth')
```
Comment (txchen) [inline test/compiler_spec.js]: This is a great change, thanks!
But is this a mistake? Why do you remove myth testing? :)

Comment (txchen) [inline test/compiler_spec.js]: Hi Steve, just saw your comment that myth is failing on your side.
Could you share more information?

Comment (txchen) [inline test/compiler_spec.js]: I figured out that it's due to myth's dependencies has been updated, transition does not need to be prefixed anymore. https://github.com/davidhund/shouldiprefix/commit/7f6aa0c8021e490bd381b8120e977264e90aa117

I will update the test case to make it pass.

**Comment by stereosteve** [comment]
@txchen thanks for the info.  Rebased on master and all is well.
@yyx990803 does this look good?

**Comment by yyx990803** [comment]
Thanks for the PR, everything looking good, except maybe update it to babel now? ;)

**Pull request merged**